lib/diagnostic.js

```diff
@@ -27,7 +30,7 @@ let access = null; // replace null with your response
 //
 //     `['alderaan', 'deathStar', 'fire', 'global']`
 //
-let scopes = [/* your response here */];
+let scopes = ['deathStar'];
```

Comment:
You're part-way right. Because `laserQuantity` is defined within the `deathStar` function, it is only accessible to things within the `deathStar` function scope. However, `fire` is defined inside `deathStar`, so its function scope is inside the `deathStar` function scope - consequently, things inside the `fire` scope can also access anything inside the `deathStar` scope.

Interestingly, that remains true even if we `return` the `fire` function and pass it out of `deathStar` -- that's the basis of closures.